editor/src/clj/editor/game_project.clj

```diff
@@ -103,8 +103,11 @@
 
 (defn- find-custom-resources [resource-map custom-paths]
   (->> (flatten (keep (fn [custom-path]
-                        (when-let [base-resource (resource-map custom-path)]
-                          (resource/resource-seq base-resource)))
+                        (let [base-resource (resource-map custom-path)]
+                          (if base-resource
+                            (resource/resource-seq base-resource)
+                            (throw (ex-info (format "Custom resource not found: '%s'" custom-path)
+                                            {})))))
```

Comment:
This is good! But you also need to catch the exception in the `custom-build-targets` output. It currently looks like this:
```clj
(output custom-build-targets g/Any :cached
        (g/fnk [_node-id resource-map settings-map]
               (let [custom-paths (parse-custom-resource-paths (get settings-map ["project" "custom_resources"]))]
                 (map (partial make-custom-build-target _node-id)
                      (find-custom-resources resource-map custom-paths)))))
```
We need to wrap the `(map ...)` expression in a try-catch and return an `ErrorValue` in case it throws an exception:
```clj
(output custom-build-targets g/Any :cached
        (g/fnk [_node-id resource-map settings-map]
               (let [custom-paths (parse-custom-resource-paths (get settings-map ["project" "custom_resources"]))]
                 (try
                   (map (partial make-custom-build-target _node-id)
                        (find-custom-resources resource-map custom-paths))
                   (catch Throwable error
                     (g/map->error
                       {:_node-id _node-id
                        :_label :custom-build-targets
                        :message (ex-message error)
                        :severity :fatal}))))))
```
`ErrorValues` are how errors are propagated in the graph. Besides the error message, they also contain information about the source `:_node-id` and `:_label` (i.e. output name) that caused the error. Here we use this to open up the `game.project` editor when the user double-clicks the resulting Build Error.

Hope this helps!

Reply:
Yes, this is exactly what I was looking for. Thanks. I didn't know about ErrorValues.